### PR TITLE
Move AST enums defined in dmd.global to astenums

### PR DESCRIPTION
**Pull request by ibuclaw**
These have nothing to do with either `struct Global` nor `struct Params` and everything to do with the AST.

### CODE REVIEW

File: src/dmd/astenums.d [hunk 1]
```diff
@@ -389,3 +391,43 @@ enum InitKind : ubyte
...
+{
+    def,      /// default
+    asStruct, /// `extern(C++, struct)`
+    asClass,  /// `extern(C++, class)`
+}
+
+/// Function match levels
+///
+/// https://dlang.org/spec/function.html#function-overloading
+enum MATCH : int
```
Comment (ljmf00) [inline src/dmd/astenums.d:419]: Can we shorten this to `byte`?

**Review by ljmf00 — APPROVED** [review]
LGTM. I left some comments, not to be done on this PR since it is unrelated, but maybe we should care.

File: src/dmd/astenums.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/astenums.d:419]: Maybe in another PR, not this though.

**Review by WalterBright — APPROVED** [review]
